### PR TITLE
Add Callout component

### PR DESCRIPTION
**Pull request by SerhiiTsybulskyi**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #50 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

Variants
Screenshot 2024-06-10 at 14 26 43

Custom Icon
Screenshot 2024-06-10 at 14 26 49

Custom Text
Screenshot 2024-06-10 at 14 26 54

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -33,7 +33,7 @@ module.exports = {
   'rules': {
     'react/display-name': 'off',
     'no-unused-vars': [0],
-    'indent': ['error', 2],
+    'indent': ['error', 2, {'SwitchCase': 1}],
```
Comment (SerhiiTsybulskyi) [inline .eslintrc.cjs:36]: Added to fix issue with index for switch/case
Screenshot 2024-06-10 at 14 30 15

File: src/layers/Callout/Callout.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  icon,
+  variant = 'default',
+  theme: customTheme
+}) => {
+  const theme = useComponentTheme('callout', customTheme);
+
+  const defaultIcon = useMemo(() => {
+    switch (variant) {
+      case 'success':
+        return <CheckCircleIcon />;
```
[the remark this answers is not in the document]
Comment (amcdnl) [reply src/layers/Callout/Callout.tsx:44]: I would probably make a new file that exends this w/ each callout theme set and icon.

File: src/layers/Callout/SuccessCallout.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import { Callout, CalloutProps } from './Callout';
+
+import CheckCircleIcon from '@/assets/icons/check_circle.svg?react';
+
+export const SuccessCallout: FC<CalloutProps> = ({ icon, text, theme }) => (
+  <Callout
+    icon={icon ?? <CheckCircleIcon />}
+    text={text}
+    variant="success"
+    theme={theme}
```
Comment (amcdnl) [inline src/layers/Callout/SuccessCallout.tsx:11]: Just spread the text/theme ----> `...rest`

**Review by amcdnl — APPROVED** [review]
Need to add docs once we release

**Comment by SerhiiTsybulskyi** [comment]
> an alternative to making a separate component for each variant could be to map the variants to their default icons and also give the ability to override that with the `icon` prop.
> 
> ```
> // Callout.tsx
> 
> const icons = {
>   default: null,
>   success: <SuccessIcon />,
>   error: <ErrorIcon />,
>   warning: <WarningIcon />,
>   info: <InfoIcon />
> }
> ```
> 
> and then render the icon like
> 
> ```
>  {icon ?? icons[variant]}
> ```

I've made solution which you described (but with switch/case for icon) first time, but Austin ask to extract it on several component to make it cleaner, that why I updated it to have several components

**Pull request merged**